src/nomad_parser_vasp/parsers/xml_parser.py

```diff
@@ -23,83 +23,88 @@
-    )
+    xml=MappingAnnotationModel(path='modeling.kpoints.varray.v[?"@name"="weights"]')
 )
+dft_path = 'modeling.separator[?"@name"="electronic exchange-correlation"]'
+# DFT.xc_functionals.m_annotations = dict(xml=MappingAnnotationModel(path=dft_path))
 DFT.exact_exchange_mixing_factor.m_annotations = dict(
     xml=MappingAnnotationModel(
         operator=(
             lambda mix, cond: mix if cond else 0,
-            [dft_path + '.i[@name="HFALPHA"]', dft_path + '.i[@name="LHFCALC"]'],
+            [dft_path + '.i[?"@name"="HFALPHA"]', dft_path + '.i[?"@name"="LHFCALC"]'],
         )
     )  # TODO convert vasp bool
 )
+XCFunctional.libxc_name.m_annotations = dict(
+    xml=MappingAnnotationModel(
+        operator=(convert_xc, [dft_path + '.i[?"@name"="GGA"]'])  # add LDA & mGGA
+    )
+)
 # ? target <structure name="initialpos" > and <structure name="finalpos" >
 AtomicCell.positions.m_annotations = dict(
-    xml=MappingAnnotationModel(path='calculation.structure.varray[@name="positions"]')
+    xml=MappingAnnotationModel(
+        path='calculation.structure.varray[?"@name"="positions"]'
+    )
 )
 """
 forces.m_annotations = dict(
-    xml=MappingAnnotationModel(path='calculation.structure.varray[@name="forces"]')
+    xml=MappingAnnotationModel(path='calculation.structure.varray[?"@name"="forces"]')
 )
 stress.m_annotations = dict(
-    xml=MappingAnnotationModel(path='calculation.structure.varray[@name="stress"]')
+    xml=MappingAnnotationModel(path='calculation.structure.varray[?"@name"="stress"]')
 )
 """
 AtomicCell.lattice_vectors.m_annotations = dict(
     xml=MappingAnnotationModel(
-        path='calculation.structure.crystal.varray[@name="basis"]'
+        path='calculation.structure.crystal.varray[?"@name"="basis"]'
     )
 )
 """
 cell_volume.m_annotations = dict(
-    xml=MappingAnnotationModel(path='calculation.structure.crystal.i[@name="volume"]')
+    xml=MappingAnnotationModel(path='calculation.structure.crystal.i[?"@name"="volume"]')
 )
 reciprocal_lattice_vectors.m_annotations = dict(
-    xml=MappingAnnotationModel(path='calculation.structure.crystal.varray[@name="rec_basis"]')
+    xml=MappingAnnotationModel(path='calculation.structure.crystal.varray[?"@name"="rec_basis"]')
 )
 total_free_energy.m_annotations = dict(
-    xml=MappingAnnotationModel(path='calculation.energy.i[@name="e_fr_energy"]')
+    xml=MappingAnnotationModel(path='calculation.energy.i[?"@name"="e_fr_energy"]')
 )
 total_internal_energy.m_annotations = dict(
-    xml=MappingAnnotationModel(path='calculation.energy.i[@name="e_0_energy"]')
+    xml=MappingAnnotationModel(path='calculation.energy.i[?"@name"="e_0_energy"]')
 )
 """
 ElectronicEigenvalues.spin_channel.m_annotations = dict(
     xml=MappingAnnotationModel(
-        path='calculation.eigenvalues.array.set.set[@comment="spin 1"]'
+        path='calculation.eigenvalues.array.set.set[?"@comment"="spin 1"]'
```

Comment:
i think in vasprun the eigenalues and occupations are in the same element so you would need to introduce an operator to manipulate the data

Reply:
Can't I  use the jmespath index? E.g.

`ElectronicEigenvalues.reciprocal_cell.m_annotations.xml + '.r[0]'`